app/uk/gov/hmrc/plasticpackagingtax/registration/models/genericregistration/Partner.scala

```diff
@@ -30,25 +30,22 @@ case class Partner(
   incorporationDetails: Option[IncorporationDetails] = None,
   partnerPartnershipDetails: Option[PartnerPartnershipDetails] = None,
   contactDetails: Option[PartnerContactDetails] = None,
-  organisationName: Option[String] =
-    None // TODO remove in favour of partner.name
+  userSuppliedName: Option[String] = None
 ) {
 
-  lazy val name: String = partnerType match {
-    case Some(PartnerTypeEnum.SOLE_TRADER) =>
-      soleTraderDetails.map(_.name).getOrElse(
-        throw new IllegalStateException("Sole Trader details absent")
-      )
-    case Some(PartnerTypeEnum.SCOTTISH_PARTNERSHIP) | Some(
-          PartnerTypeEnum.LIMITED_LIABILITY_PARTNERSHIP
-        ) | Some(PartnerTypeEnum.SCOTTISH_LIMITED_PARTNERSHIP) =>
-      partnerPartnershipDetails.flatMap(_.name).getOrElse(
-        throw new IllegalStateException("Partnership details absent")
-      )
-    case _ =>
-      incorporationDetails.map(_.companyName).getOrElse(
-        throw new IllegalStateException("Incorporation details absent")
-      )
+  lazy val name: String = {
+    val grsProvidedName = partnerType match {
+      case Some(PartnerTypeEnum.SOLE_TRADER) =>
+        soleTraderDetails.map(_.name)
+      case Some(PartnerTypeEnum.LIMITED_LIABILITY_PARTNERSHIP) | Some(
+            PartnerTypeEnum.SCOTTISH_PARTNERSHIP
+          ) | Some(PartnerTypeEnum.SCOTTISH_LIMITED_PARTNERSHIP) =>
+        partnerPartnershipDetails.flatMap(_.partnershipName)
+      case _ =>
+        incorporationDetails.map(_.companyName)
+    }
+    val availableNames = Seq(userSuppliedName, grsProvidedName).flatten
```

Comment:
User supplied name trumps GRS supplied names but at least 1 needs to be available.